.github/workflows/detect-renamed-posts.js

```diff
@@ -0,0 +1,54 @@
+const fs = require("fs");
+const path = require("path");
+const child_process = require("child_process");
+
+const redirectsFilePath = path.join(
+  path.dirname(path.dirname(__dirname)),
+  "_redirects"
+);
+
+function updateRedirects(originalName, newName) {
+  const redirects = fs.readFileSync(redirectsFilePath, "utf-8");
+  const lines = redirects.split("\n");
+
+  const newRedirect = `/blog/${originalName} /blog/${newName}`;
+
+  // search for a line with newRedirect, if not found, raise an error
+  const found = lines.find((line) => line.trim() === newRedirect.trim());
+  if (found) {
+    console.log(`Yay, found redirect for ${originalName} -> ${newName}!`);
+  } else {
+    throw new Error(
+      `Could not find a redirect for /blog/${originalName} -> /blog/${newName}`
+    );
+  }
+}
+
+// Read the changed files list from git diff. Since this runs from a GitHub
+// Actions Pull Request environment, the delta from HEAD^ is _always_ exactly
+// what we care about, example:
+//   Checking out the ref
+//     /usr/bin/git checkout --progress --force refs/remotes/pull/362/merge
+//     Note: switching to 'refs/remotes/pull/362/merge'.
+//     ...
+//     HEAD is now at 3385152 Merge aee3dbdc3da6eba401ee67de178d100ca2a20431 into c9df39b302eac7c2f79d42541f14789f40584a61
+const gitCommand = `git diff --name-status --diff-filter=R -M5 HEAD^ -- _posts/`;
+const renamedFiles = child_process
+  .execSync(gitCommand)
+  .toString()
+  .split("\n")
+  .filter((file) => file !== "");
+renamedFiles.forEach((file) => {
+  // each line should have 3 elements:
+  // R100    _posts/2020-01-01-old-post.md    _posts/2020-01-01-new-post.md
+  const fileParts = file.split("\t");
```

Comment:
My only realy concern is here. I think this is reasonable, but probably not really a contract from the git cli that these will always be tabs. That being said I don't have a better suggestion, so I think I'm just being :old_man_shakes_fist_at_javascript:

Reply:
:grimacing: oh yeah, this thing is SUPER brittle, there is undoubtedly much better libraries/options out there

Reply:
Eh, I say we send it. If it breaks it can always be disabled. I doubt this will change much in the future